cornice/resource.py

```diff
@@ -24,83 +24,127 @@ def resource(depth=1, **kw):
-        else:
-            prefixes = ('',)
-
-        for prefix in prefixes:
-
-            # get clean view arguments
-            service_args = {}
-            for k in list(kw):
-                if k.startswith('collection_'):
-                    if prefix == 'collection_':
-                        service_args[k[len(prefix):]] = kw[k]
-                elif k not in service_args:
-                    service_args[k] = kw[k]
-
-            if prefix == 'collection_' and service_args.get('collection_acl'):
-                service_args['acl'] = service_args['collection_acl']
-
-            # create service
-            service_name = (service_args.pop('name', None) or
-                            klass.__name__.lower())
-            service_name = prefix + service_name
-            service = services[service_name] = Service(name=service_name,
-                                                       depth=2, **service_args)
-
-            # initialize views
-            for verb in ('get', 'post', 'put', 'delete', 'options', 'patch'):
-
-                view_attr = prefix + verb
-                meth = getattr(klass, view_attr, None)
-
-                if meth is not None:
-                    # if the method has a __views__ arguments, then it had
-                    # been decorated by a @view decorator. get back the name of
-                    # the decorated method so we can register it properly
-                    views = getattr(meth, '__views__', [])
-                    if views:
-                        for view_args in views:
-                            service.add_view(verb, view_attr, klass=klass,
-                                             **view_args)
-                    else:
-                        service.add_view(verb, view_attr, klass=klass)
-
-        setattr(klass, '_services', services)
-
-        if VENUSIAN:
-            def callback(context, name, ob):
-                # get the callbacks registred by the inner services
-                # and call them from here when the @resource classes are being
-                # scanned by venusian.
-                for service in services.values():
-                    config = context.config.with_package(info.module)
-                    config.add_cornice_service(service)
-
-            info = venusian.attach(klass, callback, category='pyramid',
-                                   depth=depth)
-        return klass
+        return add_resource(klass, depth, **kw)
     return wrapper
 
 
+def add_resource(klass, depth=1, **kw):
+    """Function to declare resources of a Class.
+
+    All the methods of this class named by the name of HTTP resources
+    will be used as such. You can also prefix them by "collection_" and they
+    will be treated as HTTP methods for the given collection path
+    (collection_path), if any.
+
+    Here is an example::
+        class User(object):
+            pass
+
+        add_resource(User, collection_path='/users', path='/users/{id}')
```

Comment:
If possible, add the `:params:` syntax which explains what is contained in the arguments.